pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -237,17 +253,28 @@ func (g *Gatherer) updateAlertsCacheFromClient(ctx context.Context, metricsClien
 			klog.Info(logPrefix + "metric is nil")
 			continue
 		}
-
+		alert := Alert{
+			Labels: make(map[string]string),
+		}
 		for _, label := range metric.GetLabel() {
 			if label == nil {
 				klog.Info(logPrefix + "label is nil")
 				continue
 			}
-
 			if label.GetName() == "alertname" {
-				g.firingAlerts[label.GetValue()] = true
+				alert.Name = label.GetValue()
+				continue
 			}
+			alert.Labels[label.GetName()] = label.GetValue()
 		}
+		alerts, ok := g.firingAlerts[alert.Name]
+		if ok {
+			alerts = append(alerts, alert)
+		} else {
+			alerts = []Alert{alert}
+		}
+
+		g.firingAlerts[alert.Name] = alerts
```

Comment:
Could be simplified into `g.firingAlerts[alert.Name] = append(g.firingAlerts[alert.Name], alert)` :wink: